lib/mongoClient.ts

```diff
@@ -0,0 +1,81 @@
+import mongoose, { Mongoose } from "mongoose";
+
+const MONGODB_URI = process.env.MONGODB_URI;
+
+type MongooseConnectionCache = {
+  conn: Mongoose | null;
+  promise: Promise<Mongoose> | null;
+};
+
+// Encapsulate the cache inside the module-level closure, instead of polluting the global
+const cached: MongooseConnectionCache = {
+  conn: null,
+  promise: null,
+};
+
+async function dbConnect(): Promise<Mongoose> {
+  // When already have an existing connection
+  if (cached.conn) {
+    return cached.conn;
+  }
+
+  if (!cached.promise) {
+    const opts = {
+      bufferCommands: false,
+    };
+
+    if (!MONGODB_URI) {
+      throw new Error("...");
```

Comment:
Might want to specify some sort of message here. 
I checked out the pr locally and at first ran `npm run test:server` without creating an `.env.test.local` and jest crashed the test at this line.